packages/dvilela/skills/memeooorr_abci/behaviour_classes/twitter.py

```diff
@@ -320,90 +328,161 @@ def async_act(self) -> Generator:
         """Do the act, supporting asynchronous execution."""
 
         with self.context.benchmark_tool.measure(self.behaviour_id).local():
-            event = yield from self.get_event()
-
-            payload = EngageTwitterPayload(
-                sender=self.context.agent_address,
-                event=event,
-            )
+            event, new_mech_requests = yield from self.get_event()
+
+            if new_mech_requests:
+                mech_requests = json.dumps(new_mech_requests, sort_keys=True)
+                self.context.logger.info(f"Mech Requests JSON: {mech_requests}")
+
+            # add new prop to payload for tx_submitter and check for it in posttxround
+            if event == Event.MECH.value:
+                payload = EngageTwitterPayload(
+                    sender=self.context.agent_address,
+                    event=event,
+                    mech_request=json.dumps(new_mech_requests, sort_keys=True),
+                    tx_submitter=self.matching_round.auto_round_id(),
+                )
+            else:
+                payload = EngageTwitterPayload(
+                    sender=self.context.agent_address,
+                    event=event,
+                    mech_request=None,
+                    tx_submitter=self.matching_round.auto_round_id(),
+                )
 
         with self.context.benchmark_tool.measure(self.behaviour_id).consensus():
             yield from self.send_a2a_transaction(payload)
             yield from self.wait_until_round_end()
 
         self.set_done()
 
-    def get_event(self) -> Generator[None, None, str]:
+    def get_event(  # pylint: disable=too-many-statements
```

Comment:
This function is enormous now. Always try to split logic in small functions that are easy to read and mantain. It's hard to me to follow the logic. Also reduce nesting as much as possibleor you will start having a lot of corner cases you hadnt imagined.